lib/Sema/CSDiagnostics.h

```diff
@@ -479,16 +479,31 @@ class LabelingFailure final : public FailureDiagnostic {
 /// type without optional chaining or force-unwrapping it first.
 class MemberAccessOnOptionalBaseFailure final : public FailureDiagnostic {
   DeclNameRef Member;
+  Type MemberBaseType;
   bool ResultTypeIsOptional;
 
 public:
   MemberAccessOnOptionalBaseFailure(const Solution &solution,
                                     ConstraintLocator *locator,
-                                    DeclNameRef memberName, bool resultOptional)
+                                    DeclNameRef memberName,
+                                    Type memberBaseType,
+                                    bool resultOptional)
       : FailureDiagnostic(solution, locator), Member(memberName),
-        ResultTypeIsOptional(resultOptional) {}
+        MemberBaseType(memberBaseType), ResultTypeIsOptional(resultOptional) {}
 
   bool diagnoseAsError() override;
+  
+  Type getMemberBaseType() const {
+    return getSolution().simplifyType(MemberBaseType)->getRValueType();
```

Comment:
There is `resolveType` which does exactly this :) I think it would make sense to do that during construction of MemberAccessOnOptionalBaseFailure like so `MemberBaseType(resolveType(memberBaseType))`, that way this method could simply return `MemberBaseType` instead of having to do the same over multiple times...

Reply:
Great! I'll adjust, Thank you @xedin :)

Reply:
Fixed @xedin :)

Reply:
Great!